File: LeetCode/1.cpp
// USER: SIDHARTH MUDGIL
// DATE: Thursday 19, Jan 2023
// TASK: 1. Two Sum

#include <iostream>
#include <vector>
#include <map>

using namespace std;

vector<int> twoSum(vector<int> &nums, int target)
{
    int i;
    map<int, int> res;

    for (i = 0; i < nums.size(); i++)
    {
        if (res.find(target - nums[i]) != res.end())
            break;
        res[nums[i]] = i;
    }

    return {res[target - nums[i]], i};
}

int main()
{
    vector<int> nums{1, 2, 4, 6, 7};
    vector<int> res = twoSum(nums, 5);
    return 0;
}

File: leetcode/1.kt
// USER: SIDHARTH MUDGIL
// DATE: Thursday 26, Oct 2023
// PROBLEM: Two Sum

class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = mutableMapOf<Int, Int>()
        
        nums.forEachIndexed { i, v ->
            if (map.getOrDefault(target - v, -1) != -1) {
                return intArrayOf(i, map[target - v]!!)
            }

            map[v] = i
        }

        return intArrayOf()
    }
}